### ISSUE TITLE
Coco mean and variance

### ISSUE THREAD

**Issue by agirbau** [AlexeyAB/darknet#1824]
Hello!

As far as I understand, mean and variance of a dataset are computed in order to "normalize" the input images to the pretrained dataset (in this case coco I guess). The idea is to subtract the mean to the input images and divide them by variance in order to allow the network to train easily.

The question is: Does darknet do this? If yes, anyone knows where in the code? Thanks!

**Comment by AlexeyAB** [comment]
@agirbau Hi,

Is it done in batch-normalization (**mean, variance**, scale, bias)? https://github.com/AlexeyAB/darknet/blob/31df5e27356b6b11ffd43baace9afdd3800a8aa2/src/convolutional_kernels.cu#L317

that is in the 1st convolutional layer: https://github.com/AlexeyAB/darknet/blob/31df5e27356b6b11ffd43baace9afdd3800a8aa2/cfg/yolov3.cfg#L25-L26

**Comment by agirbau** [comment]
Hello!
I wasn't referring to the batch normalization. I meant that, usually, to train a net coming from a pretrained network (like imagenet or coco), the mean of the RGB channels of the dataset is subtracted to each inputting image, and then divided by the variance of the dataset. 

It's a usual practice over the research community (I believe it has something to do with training stabilization), and I was wondering if this was also done here.

**Comment by AlexeyAB** [comment]
Yolo doesn't use `Dataset normalization` because it isn't necesarry.

`Entire dataset normalization` was used when there was no `batch normalization`. When dropout was used instead of batch-normalization for regularization: https://github.com/AlexeyAB/darknet/issues/476 and https://github.com/AlexeyAB/darknet/issues/1464#issuecomment-415020511

Currently it isn't necessary to use Dataset normalization and Dropout, because we use Batch-normalization.

From your link: https://stats.stackexchange.com/a/263032/111998

> **Prior to batch normalization**, mean subtraction per channel was used to center the data around zero mean for each channel (R, G, B). This typically helps the network to learn faster since gradients act uniformly for each channel. I suspect **if you use batch normalization, the per channel mean subtraction pre-processing step is not really necessary** since you are normalizing per mini-batch anyway.

**Comment by agirbau** [comment]
Oh, I see! Thank you very much!
